### PR TITLE
tests: new test to validate refresh and revert of kernel and gadget on uc20

### PR DESCRIPTION
**Pull request by sergiocazzolato**
New test to validate refresh and revert of kernel and gadget on uc20

### CODE REVIEW

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    # The kernel and gadget snaps are refreshed
+    execute_remote "sudo snap refresh --$CORE_REFRESH_CHANNEL pc"
+    execute_remote "sudo snap refresh --$CORE_REFRESH_CHANNEL pc-kernel"
+
+    # Check the new version of the snaps is correct
+    execute_remote "snap list pc" | MATCH "^pc .* 20/$CORE_REFRESH_CHANNEL .*"
+    execute_remote "snap list pc-kernel" | MATCH "^pc-kernel .* 20/$CORE_REFRESH_CHANNEL .*"
+
+    # Reboot the system
+    execute_remote "sudo reboot"
```
Comment (bboozzoo) [inline tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:52]: should we rather wait for the system to reboot itself? preferably ask systemd whether the system is going down.

Comment (sergiocazzolato) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml]: Thanks for the review, reboot now is not manually triggered anymore

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+description: |
+    This test validates the kernel and gadget snaps can be refreshed
+    and reverted to the new snaps published to edge channel.
+
+systems: [ubuntu-20.04-*]
+
+environment:
+    CORE_CHANNEL: beta
+    CORE_REFRESH_CHANNEL: edge
```
Comment (anonymouse64) [inline tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:11]: these aren't channels for the core snap, they are channels for the gadget or kernel snaps, so could we rename these to just SNAP_CHANNEL and SNAP_REFRESH_CHANNEL?

```suggestion
    SNAP_CHANNEL: beta
    SNAP_REFRESH_CHANNEL: edge
```

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    BUILD_SNAPD_FROM_CURRENT: false
+    USE_CLOUD_INIT: true
+    ENABLE_SECURE_BOOT: true
+    ENABLE_TPM: true
+    SNAP/kernel: pc-kernel
+    SNAP/gadget: pc
```
Comment (anonymouse64) [inline tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:17]: I think we also want these tests for snapd and core20 too, no?

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:11]: Thanks for hte review, in this case CORE_CHANNEL is already deifned in nested.sh and is the variable used to create the image, which was created when we just had core16. 
I agree that the name is not representative anymore, but as it requires many changes on other tests and spread.yaml I would prefer to make is in a folowing PR. 
Is it ok for you?

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:17]: It was requested just to make it for those snaps, but it should be really easy to extend it to snapd and core20. @pedronis is it ok for you?

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml (hunk 2 above)
Comment (anonymouse64) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:11]: ah yes that's fine then I didn't realize that this var was needed/used in multiple places

File: tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml (hunk 3 above)
Comment (pedronis) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:17]: yes, though not blockers, of the two snapd is higher priority, while core20 for sure could be done as a follow up

Comment (sergiocazzolato) [reply tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml:17]: @pedronis nice, thanks

**Comment by sergiocazzolato** [comment]
@anonymouse64 i'll create a follow up PR adding the new snaps to the test.

**Pull request merged**